src/physics/mod.rs

```diff
@@ -4,12 +4,148 @@ pub mod rapier;
 #[cfg(feature = "avian")]
 pub mod avian;
 
-use bevy::prelude::*;
+use std::sync::Arc;
+
+use bevy::{ecs::system::EntityCommands, prelude::*};
 use bevy_ecs_tilemap::prelude::*;
 use tiled::{ObjectData, ObjectLayerData};
 
 use crate::prelude::*;
 
+/// Trait defining a generic way of handling colliders
+/// across different physics backends.
+pub trait HandleColliders {
+    /// Load shapes from an object layer as physics colliders.
+    ///
+    /// By default `bevy_ecs_tiled` will only process object layers
+    /// named in `collision_layer_names` in `TiledMapSettings`,
+    /// and tileset collision shapes named in `collision_object_names`.
+    ///
+    /// Collision layer names are case-insensitive and leading/trailing
+    /// whitespace is stripped out.
+    fn insert_colliders_from_shapes(
+        &self,
+        parent_entity: Entity,
+        map_type: &TilemapType,
+        grid_size: Option<&TilemapGridSize>,
+        object_data: &ObjectData,
+    ) -> EntityCommands;
+}
+
+#[derive(Clone, Resource)]
```

Comment:
> What about a new field in TiledMapSettings struct instead ?

This makes sense, I think. It does open up a weird possibility of having multiple `TiledMapBundle`s spawned, all with different physics backends, and you have to be explicit to set the physics backend each time you spawn them in your code. I'm not sure these are reasons not to do it. My original thought behind making it a `Resource` is that it seemed like the physics backend should be a global configuration.

Reply:
Yeah, good point!

Agree with your observation, but if for an obscure reason the user wants to do so, why not let him ? I feel we should always give user the choice.
However, I agree with you, it adds a bit of boiler plate code.... And more concerning, it forces the user to update a working code to add this boiler plate. It's not really a breaking change (it's close) but it's definitivelly not "dev-friendly".

I'm still not fond of using a `Resource`.
Right now, we have everything inside the `TiledMapBundle` and I would like to keep it this way if possible: it allows to have different maps with different settings and not rely on global settings. Again, give user the choice to do as he likes.
Moreover, having a `Resource` will not solve the need to update a working code.

One idea to solve this: implement `Default` trait for the `PhysicsBackend` and make the default value based upon which feature is enabled:
- if avian feature is enabled, make the default be `Avian`
- if rapier feature is enabled, make the default be `Rapier`
- if none are enabled, make the default be `None`

It feels a bit "hacky" but I cannot find a good point against it on the Internet. We would not change the enum but just change its default value.

I believe it would solve our issues:
- no boiler plate in most cases
- no need to update existing code (and bonus point: no need to update examples :) )
- if he wishes to, user can still have control over which physic backend he wants to use in which case


What do you think ?

Reply:
> One idea to solve this: implement Default trait for the PhysicsBackend and make the default value based upon which feature is enabled:

I think I like this.